tests/test_rescue.py

```diff
@@ -10,569 +10,447 @@
+        rescue_plain_fx.marked_for_deletion = 'This is not a valid reason'
+
+    with pytest.raises(TypeError):
+        rescue_plain_fx.marked_for_deletion = {'Reason 123': 'Bowl of Petunias'}
+
+
+def test_rescue_lang_id_constructor(rescue_sop_fx):
+    """
+    Verifies Language_ID is returned
+    """
+    # Defaults to EN
+    assert rescue_sop_fx.lang_id == 'EN'
+
+
+@pytest.mark.parametrize("garbage", [None, 42, -2.2, uuid4()])
+def test_lang_id_garbage(garbage, rescue_plain_fx: Rescue):
+    """
+    Verifies throwing garbage types at Rescue.lang_id results in a TypeError
+    Args:
+        garbage (): Garbage to throw
+        rescue_plain_fx (Rescue): Plain rescue Fixture
+    """
+    with pytest.raises(TypeError):
+        rescue_plain_fx.lang_id = garbage
+
+
+def test_platform_raises_type_error(rescue_sop_fx):
+    """
+    Verifies that Rescue.Platform properly raises a TypeError when the
+    wrong value is supplied.
+    """
+    with pytest.raises(TypeError):
+        rescue_sop_fx.platform = 'Xbox'
+
+
+@pytest.mark.asyncio
+async def test_add_rats_bad_id(rat_no_id_fx, rescue_sop_fx):
+    """
+    Verifies attempting to add a rat that does not have a API id fails as expected
+    """
+    with pytest.raises(ValueError, message="Assigned rat does not have a known API ID"):
+        await rescue_sop_fx.add_rat(rat=rat_no_id_fx)
         assert rat_no_id_fx not in rescue_sop_fx.rats
 
-    @pytest.mark.asyncio
-    async def test_add_rats_ok(self, rat_good_fx, rescue_sop_fx):
-        """
+
+@pytest.mark.asyncio
+async def test_add_rats_ok(rat_good_fx, rescue_sop_fx):
+    """
         Verifies adding a existing rat with a UUID works
         Args:
             rat_good_fx (Rats): Good Rat object Test Fixture
             rescue_sop_fx (Rescue):  Rescue object Test Fixture
-        """
-        # rescue_sop_fx:Rescue
-        await rescue_sop_fx.add_rat(rat=rat_good_fx)
-        assert rat_good_fx in rescue_sop_fx.rats
-
-    @pytest.mark.asyncio
-    async def test_add_rat_from_cache(self, rat_good_fx: Rats, rescue_sop_fx: Rescue):
-        await rescue_sop_fx.add_rat(rat_good_fx.name)
-        assert rat_good_fx == rescue_sop_fx.rats[0]
-
-    @pytest.mark.parametrize("garbage", [(None,), (42,), (-2.2,), (uuid4(),)])
-    def test_irc_nickname_garbage(self, garbage, rescue_plain_fx: Rescue):
-        """
-        Verifies throwing garbage types at Rescue.irc_nickname results in a TypeError
-        Args:
+    """
+    # rescue_sop_fx:Rescue
+    await rescue_sop_fx.add_rat(rat=rat_good_fx)
+    assert rat_good_fx in rescue_sop_fx.rats
+
+
+@pytest.mark.asyncio
+async def test_add_rat_from_cache(rat_good_fx: Rats, rescue_sop_fx: Rescue):
+    """
+    Verifies rat added from cache matches rat object in Rescue
+    """
+    await rescue_sop_fx.add_rat(rat_good_fx.name)
+    assert rat_good_fx == rescue_sop_fx.rats[0]
+
+
+@pytest.mark.parametrize("garbage", [(None,), (42,), (-2.2,), (uuid4(),)])
```

Comment:
should this not just be a plain list of objects rather than a list of single-element tupples? this test otherwise only tests tupples of len 1

Reply:
```py
@pytest.mark.parametrize("garbage", [(None), (42), (-2.2), (uuid4())])
```
Changed to proper format for multiple tests.  Pycharm is whining about extraneous parentheses, but the syntax is valid.